src/hotspot/cpu/aarch64/c1_MacroAssembler_aarch64.cpp

```diff
@@ -101,7 +101,8 @@ void C1_MacroAssembler::try_allocate(Register obj, Register var_size_in_bytes, i
 void C1_MacroAssembler::initialize_header(Register obj, Register klass, Register len, Register t1, Register t2) {
   assert_different_registers(obj, klass, len);
 
-  if (UseCompactObjectHeaders || EnableValhalla) {
+  // EnableValhalla legacy
```

Comment:
I don't think this comment is necessary and kind of noisy everywhere, especially since there's Arguments::enable_preview() also tested.  The weird thing is that EnableValhalla is true but Arguments::enable_preview() is not always true so I see why it's good to have testing with both settings, we were exercising valhalla code even with enable-preview off.